### PR TITLE
Add several spelling corrections + refinements

### PR DESCRIPTION
**Pull request by luzpaz**

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -3454,6 +3459,7 @@ coctail->cocktail
 cocument->document
 cocumentation->documentation
 cocuments->document
+codde->code, coded,
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Suggest coddle

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -5958,6 +5969,7 @@ develoments->developments
 develope->develop
 developement->development
 developements->developments
+developme->development
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Suggest `develop me`

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -6167,6 +6179,10 @@ diffrerences->differences
 diffult->difficult
 dificulties->difficulties
 dificulty->difficulty
+difine->define
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Suggest `divine`

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -6167,6 +6179,10 @@ diffrerences->differences
...
+difined->defined
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Likewise

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -6167,6 +6179,10 @@ diffrerences->differences
...
+difines->defines
+difining->defining
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Likewise

File: codespell_lib/data/dictionary.txt [hunk 6]
```diff
@@ -14320,6 +14358,7 @@ programers->programmers
 programing->programming
 programm->program
 programmaticaly->programmatically
+programme->program, programmed, programmer,
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Remove, this is valid BE:
https://dictionary.cambridge.org/dictionary/english/programme

File: codespell_lib/data/dictionary.txt [hunk 7]
```diff
@@ -16112,6 +16153,10 @@ sclar->scalar
 scoll->scroll
 scolling->scrolling
 scopeing->scoping
+scrach->scratch
+scrached->scratched
+scraches->scratches
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt]: Possibly add `scrachs->scratches`

**Comment by luzpaz** [comment]
Integrated changes. Please re-review

**Pull request merged**